test/v4/controllers/RetrieveCalculationControllerSpec.scala

```diff
@@ -165,19 +189,37 @@ class RetrieveCalculationControllerSpec
     }
   }
 
-  private trait Test extends ControllerTest {
+  private trait Test extends ControllerTest with AuditEventChecking {
 
-    private lazy val controller = new RetrieveCalculationController(
+    lazy val controller = new RetrieveCalculationController(
       authService = mockEnrolmentsAuthService,
       lookupService = mockMtdIdLookupService,
       validatorFactory = mockRetrieveCalculationValidatorFactory,
       service = mockRetrieveCalculationService,
       cc = cc,
       hateoasFactory = mockHateoasFactory,
-      idGenerator = mockIdGenerator
+      idGenerator = mockIdGenerator,
+      auditService = mockAuditService
     )
 
-    protected def callController(): Future[Result] = controller.retrieveCalculation(nino, taxYear, calculationId)(fakeRequest)
+    protected def callController(): Future[Result] =
+      controller.retrieveCalculation(nino, taxYear, calculationId)(fakeRequest)
+
+    protected def event(auditResponse: AuditResponse, requestBody: Option[JsValue] = None): AuditEvent[GenericAuditDetail] =
+      AuditEvent(
+        auditType = "RetrieveATaxCalculation",
+        transactionName = "retrieve-a-tax-calculation",
+        detail = GenericAuditDetail(
+          versionNumber = "4.0",
```

Comment:
We can more flexibly expect the version to be whatever is in the request, and make it clearer where the "4.0" is coming from:
```suggestion
          versionNumber = apiVersion.name,
```